tests/messages_tests/test_cookie.py

```diff
@@ -116,15 +120,20 @@ def test_max_cookie_length(self):
         # size which will fit 4 messages into the cookie, but not 5.
         # See also FallbackTest.test_session_fallback
         msg_size = int((CookieStorage.max_cookie_size - 54) / 4.5 - 37)
+        first_msg = None
         for i in range(5):
-            storage.add(constants.INFO, str(i) * msg_size)
+            # TODO: Can we fix the randomness here somehow? This test might get flacky :/
+            s = str(i) + ''.join(random.choice(string.ascii_letters) for _ in range(msg_size - 1))
```

Comment:
> since the input will always end up being a string of ascii letters.

Yes, but with a bit of luck a string that zlib could compress; hence hardcoding or fixing the rng seed would ensure that the test stays stable.

Reply:
I see what you mean - with tests being run so often it's only a matter of time.
To resolve it we can call `random.seed(seed + i)` in the for loop before the string is created. And set `seed = 42`(or some other int) outside the loop. This way we'll have a set of known good random strings that aren't the same. Since if they are the same the zlib compression will detect the duplication and compress them efficiently, since the messages are stored in a single cookie which is processed all at once. BTW I tested with the initial `seed =42` and it passes the messages tests.

Reply:
There is no need to reseed with every iteration; just one before the loop should be enough. I also think that it is probably not that likely that zlib will compress that string by much. But maybe it is better to be safe than sorry.

Do you think you could finish those two pull requests?

Reply:
Happily - I will get onto it today.

Reply:
New PR #13800 with tests